### DISCUSSION TITLE
Is there a way to change error handler for single endpoint?

### DISCUSSION THREAD

**Discussion by ghost** [vitalik/django-ninja#1325] [Q&A] [answered]
Hello,

I'm wondering if there is a way to change error handler for ValidationError for just single endpoint. I'm using Pydantic's model validator to check if user exists in database to keep the error messages consistent and my endpoint should return 401 status code instead of standard 422.

**Comment by vitalik** [comment, marked as answer]
Hi @flisakl 

basically there are two options:


1) define [custom exception handler](https://django-ninja.dev/guides/errors/#custom-exception-handlers)

```Python
class MyException(Exception):
    pass

@api.exception_handler(MyException)
def custom_401_handler(request, exc):
    return api.create_response(
        request,
        {"message": "Invalid credentials or something"},
        status=401,
    )

@api.post("/some")
def some_view(request):
    if something:  # Example: user does not exist
        raise MyException("User not found")
    return {"detail": "User exists"}
```

2) do it in operations and define in docs:

```Python
@api.post("/some", response={200: SomeSchema, 401: OtherSchema}
def some(request):
    if some:
        return 401, {"some": "detail"}
   return {"normal": "response"}
```

**Reply by ghost** [reply]
Thanks for the reply. With first option I've managed to solve my problem.
